ml-prims/src/metrics/metrics.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "linalg/eltwise.h"
+#include "stats/mean.h"
+#include "linalg/subtract.h"
+#include "linalg/power.h"
+
+#include <thrust/reduce.h>
+#include <thrust/device_ptr.h>
+
+namespace MLCommon {
+    namespace Metrics {
+        template<typename math_t>
+        math_t r_squared(math_t *y, math_t *y_hat, int n) {
```

Comment:
This is a simpe function, but might be good practice to start including docstrings here.